### PR TITLE
Fix Special Tests

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

There is a failing special test on master due to some of the port issues. I noticed that this seemed to be related to running two tests directly after each other for spawn/normal sharded ddp, thus split them out and added a todo. I also removed a hard-coded port setting.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by tchaton** [comment]
Hey @SeanNaren,

Any updates on this one ?

Best,
T.C

File: tests/plugins/test_sharded_plugin.py [hunk 1]
```diff
@@ -297,12 +297,24 @@ def training_step(self, batch, batch_idx):
 
 
 @RunIf(skip_windows=True, special=True, fairscale=True, min_gpus=2)
-@pytest.mark.parametrize("accelerator", ["ddp_sharded", "ddp_sharded_spawn"])
-def test_ddp_sharded_plugin_manual_optimization(tmpdir, accelerator):
+def test_ddp_sharded_plugin_manual_optimization_spawn(tmpdir):
+    # todo (sean): this test has been split out as running both tests using parametrize causes "Address in use"
```
Comment (kaushikb11) [inline tests/plugins/test_sharded_plugin.py:301]: @SeanNaren Couldn't we just pass a different `MASTER_PORT` env variable with `accelerator` using parameterize?

Comment (SeanNaren) [reply tests/plugins/test_sharded_plugin.py:301]: good idea! lemme try it now

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by SeanNaren** [comment]
Let's merge as is to get the special tests working. I'm having difficulty getting it to work with random ports for some weird whacky reason...

**Pull request merged**